tests/test_snapshots.py

```diff
@@ -0,0 +1,39 @@
+from pysages.backends.snapshot import Box, Snapshot
+from pysages.utils import ToCPU, copy
+
+import jax.numpy as np
+import numpy
+import pytest
+
+
+def test_copying():
+    positions = np.ones((2, 3))
+    vel_mass = np.zeros((2, 4))
+    forces = np.zeros((2, 3))
+    ids = np.zeros((2,), dtype = np.uint32)
+    H = (
+        (1.0, 0.0, 0.0),
+        (0.0, 1.0, 0.0),
+        (0.0, 0.0, 1.0),
+    )
+    origin = (0.0, 0.0, 0.0)
+    box = Box(H, origin)
+    dt = 0.1
+
+    old = Snapshot(positions, vel_mass, forces, ids, box, dt)
+    new = copy(old)
+
+    old_ptr = old.positions.unsafe_buffer_pointer()
+    new_ptr = new.positions.unsafe_buffer_pointer()
+    old_box_H_ptr = old.box.H.unsafe_buffer_pointer()
+    new_box_H_ptr = new.box.H.unsafe_buffer_pointer()
+
+    # When copying to CPU we get a `numpy.ndarray` instead of a
+    # `jaxlib.xla_extension.DeviceArray`
+    new_cpu = copy(old, ToCPU())
+
+    assert np.all(old.positions == new.positions).item()
```

Comment:
Ok, I was double checking if assert is the right way to test in automated frameworks.
But it seems like that pytest relies on assert, while unittest wants exceptions.

Reply:
Do you have a preference on `unittest`. I like that `pytest` doesn't require too much boilerplate.

Reply:
no, I don't have a preference. My testing experience is more about C++. I read up on it and seems like it is as you say. both are fine and pytest is easier to work with. So I am happy to go along with it.